cmd/twosum/main_test.go

```diff
@@ -55,14 +67,54 @@ func Test_doubleSumMap(t *testing.T) {
 	}
 }
 
+func getTestData() *TestData {
+	once := &sync.Once{}
+	once.Do(func() {
+		m := make(map[int]bool)
+		sums := make(map[int]bool)
+		nums := make([]int, 2000)
+		var target int
+		r := rand.New(rand.NewSource(time.Now().Unix()))
+		for len(m) < 2000 {
+			m[r.Intn(3000)] = true
+		}
+		i := 0
+		for key, _ := range m {
+			nums[i] = key
+			i++
+		}
+		for i := 0; i < len(nums)-1; i++ {
+			for j := i + 1; j < len(nums); j++ {
+				if _, exist := sums[nums[i]+nums[j]]; exist {
+					sums[nums[i]+nums[j]] = false
```

Comment:
if you need `true` value only below then probably no need to assign `false` here

Reply:
furthermore, as long as you don't use `sums` you can skip it completely and make `target = nums[len(nums-2]+nums[len(nums-1]` for instance

Reply:
i try to keep problem condition about only one solution in input slice, but if i use las two elements in slice this can be righ only if my slice is sorted

Reply:
ah, makes sense then :)

Reply:
as long as you use `if _, exist := sums[nums[i]+nums[j]]; exist` instead of `if sums[nums[i]+nums[j]]` you could save some memory with `map[int]struct{}` :)